plutus-core/plutus-core/src/PlutusCore/Pretty/Readable.hs

```diff
@@ -51,19 +56,38 @@ type HasPrettyConfigReadable env configName =
 
 makeLenses ''PrettyConfigReadable
 
-instance configName ~ PrettyConfigName => HasPrettyConfigName (PrettyConfigReadable configName) where
+instance configName ~ PrettyConfigName =>
+        HasPrettyConfigName (PrettyConfigReadable configName) where
     toPrettyConfigName = _pcrConfigName
 
 instance HasRenderContext (PrettyConfigReadable configName) where
     renderContext = pcrRenderContext
 
--- | The fixity of a binder.
-binderFixity :: Fixity
-binderFixity = Fixity RightAssociative 1
+-- | For rendering things in a readable manner regardless of the pretty-printing function chosen.
+-- I.e. all of 'show', 'pretty', 'prettyClassicDef' will use 'PrettyReadable' instead of doing what
+-- they normally do. @prettyBy config (AsReadable x)@ requires @config@ to have a 'PrettyConfigName'
+-- and respects it.
+--
+-- This wrapper can be particularly useful if you want to apply a function having a 'Show' or
+-- 'Pretty' or 'PrettyClassic' or 'PrettyPlc' or whatever constraint, but want to get the argument
+-- rendered in a readable manner instead.
+newtype AsReadable a = AsReadable
+    { unAsReadable :: a
+    }
 
--- | The fixity of @(->)@.
-arrowFixity :: Fixity
-arrowFixity = Fixity RightAssociative 2
+instance (HasPrettyConfigName config, PrettyReadable a) =>
+        DefaultPrettyBy config (AsReadable a) where
+    defaultPrettyBy config (AsReadable x) =
+        prettyBy (topPrettyConfigReadable (toPrettyConfigName config) def) x
```

Comment:
@effectfully What's the reason to use `topPrettyConfigReadable` instead of `botPrettyConfigReadable` here? The `top` one wraps the entire thing in a pair of parentheses which seems unnecessary.

Reply:
I thought it was "safer", but now that you mention it, I probably just made the wrong choice. Feel free to change it.